docs/adr/0006-2021-05-26-authorisation.md

```diff
@@ -0,0 +1,92 @@
+# Authorisation of access to the Beacons Service
+
+## Status
+
+Proposed
+
+## Context
+
+- The [Beacons Service](https://github.com/mcagov/beacons-service) provides sensitive user data, and as
+  such only authorised clients should be able to access it.
+- The Maritime & Coastguard Agency requires that [Azure Active Directory](https://azure.microsoft.com/en-gb/services/active-directory/) is used as the common authorisation
+  and identity provider for its internal, "back office" services.
+- The Maritime & Coastguard Agency requires that [Azure AD B2C](https://azure.microsoft.com/en-gb/services/active-directory/) is used as the common authorisation and identity
+  provider for its external, "consumer-facing" services.
+
+(Note that Azure AD is a distinct product from Azure AD B2C and on-prem Active Directory for Windows Server. More info
+[here](https://www.predicagroup.com/blog/azure-ad-b2b-b2c-puzzled-out/).)
+
+## Decision
+
+_What is the change that we're proposing and/or doing?_
+
+We will implement a blended authorisation architecture for the Beacons domain using Azure B2C for public-facing
+accounts, Azure AD for internal accounts with role-based access control, and application code.
+
+1.  End-users (members of the public: beacon owners, account holders) will authenticate using Azure B2C to
+    access the consumer-facing [Web App](https://github.com/mcagov/beacons-webapp).
+2.  The Web App will authenticate with Azure AD as a [confidential client application](https://docs.microsoft.com/en-us/azure/healthcare-apis/fhir/register-confidential-azure-ad-client-app) to access the
+    Beacons [Service API](https://github.com/mcagov/beacons-service) with scope to perform CRUD operations on all
+    beacon records.
+3.  Application code in the Web App will restrict the permissions of the current Beacon Owner / public account
+    holder to operate only on beacon records within their scope.
+4.  Maritime & Coastguard Agency internal users will authenticate using Azure AD to access the Beacons [Back Office
+    Admin Tool](https://github.com/mcagov/beacons-backoffice). The Maritime & Coastguard Agency's production tenant
+    will be used to allow Single Sign-On. Using Azure AD will allow role-based access control of back office
+    operations, giving the ability to make Search & Rescue users 'read only' while allowing Beacon Registration Team
+    users to add flags, comments and other metadata.
+
+![Beacons authorisation diagram](assets/beacons-auth-diagram.png).
```

Comment:
Is this a png extension?  Not sure if that will open up the excalidraw picture

Reply:
This is correct.  There's a .png in there for display in the ADR and the .excalidraw file is included if we want to use it for a future ADR / update it.

Reply:
Ah yes, sorry missed that